### PR TITLE
samples: keys: Fix HW unique key test configuration 

### PR DESCRIPTION
**Pull request by joerchan**
Fix random_hw_unique_key sample tag to match sample name.
A sample with name hw_unique_key exists.

This commit adds support for nRF21540 Front-End Module to
the HUK sample doc. Adding because it was added to the Random HUK
sample.
Ref NCSDK-10897

### CODE REVIEW

File: samples/keys/random_hw_unique_key/sample.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ sample:
   description: Generate and write random keys to the KMU.
   name: Random HW Unique Key
 common:
-    tags: hw_unique_key
+    tags: random_hw_unique_key
```
Comment (oyvindronningstad) [inline samples/keys/random_hw_unique_key/sample.yaml:5]: Maybe leave both of them. The tag can be used to run all HUK samples with Twister.

**Review by peknis — APPROVED** [review]

**Review by PerMac — CHANGES_REQUESTED** [review]
tl,dr: Can you please also fix the test scenario name here https://github.com/nrfconnect/sdk-nrf/blob/main/samples/keys/random_hw_unique_key/sample.yaml#L19? 

Interesting coincidence:) I was just recently reported by @torsteingrindvik that twister has issues with these tests under some circumstances:
Twister requires unique names of scenarios within a single yaml, since test item identification is done by path+scenario_name.  Path is taken in relative to the place where twister started to look for tests. 
Scenarios' names are the same for both of this samples. Therefore, such call from `sample/keys` folder
`twister --build-only --clobber-output -T hw_unique_key/ -T random_hw_unique_key/ --outdir build-rc2`
will make twister to overwrite the same test item and report only a single result. This does not happen if called from other place, but can cause troubles in the above scenario.

File: samples/keys/random_hw_unique_key/sample.yaml (hunk 1 above)
Comment (torsteingrindvik) [reply samples/keys/random_hw_unique_key/sample.yaml:5]: Concerning tags: I recently pushed some changes to tags: https://github.com/nrfconnect/sdk-nrf/blob/main/tests/crypto/testcase.yaml

I'm starting to use a scheme where I use a common tag for things which should be able to be grouped in CI (here: `legacy`).

Maybe use something like `huk` as a tag if you need to do this as well?
There's always `--testcase-root samples/keys` in order to include both without tags as well.

Maybe @PerMac  has thoughts on best practice here

Comment (PerMac) [reply samples/keys/random_hw_unique_key/sample.yaml:5]: We don't have any defined schema for tags nor any list of used tags. Maybe at one point we will do yamls evaluation and propose some schema, but right now there are no best practices, sorry.

File: samples/keys/hw_unique_key/sample.yaml [hunk 2]
```diff
@@ -2,14 +2,15 @@ sample:
   description: Generate and write random keys to the KMU.
   name: Random HW Unique Key
 common:
-    tags: hw_unique_key
-    platform_allow: nrf5340dk_nrf5340_cpuapp nrf5340dk_nrf5340_cpuapp_ns nrf9160dk_nrf9160 nrf9160dk_nrf9160_ns nrf52840dk_nrf52840
+    tags: huk
```
Comment (oyvindronningstad) [inline samples/keys/hw_unique_key/sample.yaml:5]: The `hw_unique_key` tag is used in `tests/lib/hw_unique_key` and `tests/lib/hw_unique_key_tfm` as well. Please update those as well.

**Review by oyvindronningstad — CHANGES_REQUESTED** [review]

File: samples/keys/hw_unique_key/sample.yaml (hunk 2 above)
Comment (joerchan) [reply samples/keys/hw_unique_key/sample.yaml:5]: Thanks. Updated in testcase.yaml files as well now.

**Review by oyvindronningstad — APPROVED** [review]

**Pull request merged**